onyo/lib/onyo.py

```diff
@@ -20,12 +20,29 @@ class OnyoProtectedPathError(Exception):
 class Repo:
     """
     """
-    def __init__(self, path: Union[Path, str], init: bool = False) -> None:
+    def __init__(self, path: Union[Path, str], find_root: bool = False,
+                 init: bool = False) -> None:
+        """
+        Instantiates a `Repo` object with `path` as the root directory.
+
+        If `find_root=True` searches the root of a repository from `path`.
+
+        If `init=True`, the `path` will be initialized as a git repo and a
+        `.onyo/` directory will be created.
+        Otherwise the validity of the onyo repository is verified, and if the
+        path is invalid a `OnyoInvalidRepoError` is raised.
+        """
+        path = Path(path)
+        path = self._find_root(path) if find_root else path.resolve()
+
         if init:
             self._init(path)
+        elif not Repo._is_onyo_root(path):
+            log.error(f"'{path}' is no valid Onyo Repository.")
+            raise OnyoInvalidRepoError(f"'{path}' is no valid Onyo Repository.")
```

Comment:
Change the order of treating `init` and `find_root`?
The issue is, that there's no point in failing on `Repo('some/path', init=True, find_root=True)`, because `find_root` doesn't find anything. That's actually fine.

```suggestion
        path = Path(path)
        
        if init:
            path = path.resolve()
            self._init(path)
        else:        
            path = self._find_root(path) if find_root else path.resolve()
            if not Repo._is_onyo_root(path):
                log.error(f"'{path}' is no valid Onyo Repository.")
                raise OnyoInvalidRepoError(f"'{path}' is no valid Onyo Repository.")
```

Reply:
Thanks! I added your suggestion, am currently running the tests locally, and then will push again! :)